### PR TITLE
client: apply engine api changes for devnet 8

### PR DESCRIPTION
**Pull request by g11tech**
pending tasks to make client devnet 8 compatible

 - fcu v3
 - change validations of newpayload v3 and make them 1-1

Could have further changes depending on the acceptance of 
  - https://github.com/ethereum/consensus-specs/pull/3454

### CODE REVIEW

**Review by acolytec3 — CHANGES_REQUESTED** [review]
Looks good!  I think one error message needs to be updated but will let you confirm.

File: packages/client/src/rpc/modules/engine.ts [hunk 1]
```diff
@@ -741,14 +763,31 @@ export class Engine {
   }
 
   async newPayloadV1(params: [ExecutionPayloadV1]): Promise<PayloadStatusV1> {
+    const shanghaiTimestamp = this.chain.config.chainCommon.hardforkTimestamp(Hardfork.Shanghai)
+    const ts = parseInt(params[0].timestamp)
+    if (shanghaiTimestamp !== null && ts >= shanghaiTimestamp) {
+      throw {
+        code: INVALID_PARAMS,
+        message: 'NewPayloadV2 MUST be used after Cancun is activated',
```
[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/client/src/rpc/modules/engine.ts:776]: Any comment on this error message adjustment?

[the remark this answers is not in the document]
Comment (g11tech) [reply packages/client/src/rpc/modules/engine.ts]: thanks @acolytec3 , resolved

**Comment by g11tech** [comment]
> Looks good! I think one error message needs to be updated but will let you confirm.

thank yea updated, gtg for merge now :)

File: packages/client/test/rpc/engine/newPayloadV3.spec.ts [hunk 2]
```diff
@@ -27,276 +25,71 @@ export const batchBlocks = async (server: HttpServer) => {
     await baseRequest(server, req, 200, expectRes, false, false)
   }
 }
+const parentBeaconBlockRoot = '0x42942949c4ed512cd85c2cb54ca88591338cbb0564d3a2bea7961a639ef29d64'
 
 describe(`${method}: call with executionPayloadV1`, () => {
-  it('call with invalid block hash without 0x', async () => {
```
Comment (g11tech) [inline packages/client/test/rpc/engine/newPayloadV3.spec.ts:32]: these tests which are repeat of older versions have been removed

**Comment by holgerd77** [comment]
Client tests failing, is this this timeout thing?

**Comment by g11tech** [comment]
> Client tests failing, is this this timeout thing?

yea seems like it :slightly_smiling_face:

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**